stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/listener/SendMessageListenerImpl.kt

```diff
@@ -0,0 +1,47 @@
+package io.getstream.chat.android.offline.experimental.plugin.listener
+
+import android.content.Context
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.experimental.plugin.listeners.SendMessageListener
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.offline.experimental.global.GlobalState
+import io.getstream.chat.android.offline.experimental.plugin.logic.LogicRegistry
+import io.getstream.chat.android.offline.message.MessageSendingServiceFactory
+import io.getstream.chat.android.offline.repository.RepositoryFacade
+import kotlinx.coroutines.CoroutineScope
+
+@ExperimentalStreamChatApi
+internal class SendMessageListenerImpl(
+    private val context: Context,
+    private val logic: LogicRegistry,
+    private val globalState: GlobalState,
+    private val scope: CoroutineScope,
+    private val repos: RepositoryFacade,
+) : SendMessageListener {
+
+    override suspend fun onMessageSendResult(
+        result: Result<Message>,
+        channelType: String,
+        channelId: String,
+        message: Message,
+    ) {
+        val service = MessageSendingServiceFactory.getOrCreateService(
+            logic,
+            globalState,
+            channelType,
+            channelId,
+            scope,
+            repos,
+            context,
+            ChatClient.instance()
+        )
+
+        if (result.isSuccess) {
+            service.handleSendMessageSuccess(result.data())
```

Comment:
This is already happening here: 
```
        chatClient.channel(message.cid).sendMessageInternal(message)
            .retry(scope, chatClient.retryPolicy).await()
            .mapSuspend(::handleSendMessageSuccess)
            .recoverSuspend { error -> handleSendMessageFail(message, error) }
```
 We could simply use the `chatClient.channel(message.cid).sendMessage(message)` and remove the `handleSendMessageSuccess` and `handleSendMessageFail`

Reply:
`ChannelClient::sendMessageInternal` inside `doSend` is used only by `ChatDomain`. We can't directly use `ChannelClient::sendMessage` with ChatDomain. We need to delete `ChatDomain::sendMessage`.